DQM/GEM/plugins/GEMEfficiencyAnalyzer.cc

```diff
@@ -558,21 +567,29 @@ GEMEfficiencyAnalyzer::StartingState GEMEfficiencyAnalyzer::buildStartingState(
     }
   }
 
-  found &= state.isValid();
-
-  if (MuonHitHelper::isGEM(det_id)) {
-    const GEMDetId start_id{det_id};
+  if (found) {
+    found &= state.isValid();
+  }
 
-    const bool are_same_region = gem_layer.id.region() == start_id.region();
-    const bool are_same_station = gem_layer.id.station() == start_id.station();
-    const bool are_same_layer = gem_layer.id.layer() == start_id.layer();
-    if (are_same_region and are_same_station and are_same_layer) {
-      LogDebug(kLogCategory_)
-          << "The starting detector of the muon propagation is same with the destination. Skip this propagation.";
-      found = false;
-    }
+  if (found and (det_id.det() == DetId::Detector::Muon)) {
```

Comment:
Is it acceptable that `found` remains `true` is the detector is not `Muon`?

Reply:
When the analyzer uses tracker muons, `det_id.det()` will returns `DetId::Detector::Tracker`. Also, when the analyzer uses global muons and builds an initial state from `InnermostMeasurementState`, the same thing happens.